client/src/js/references/components/Import.js

```diff
@@ -80,6 +81,8 @@ class ImportReference extends React.Component {
     };
 
     render() {
+        this.state.onChange = this.handleChange;
+        this.state.toggle = this.toggleCheck;
```

Comment:
Do not mutate state in `render()` method. You must use `setState()`. State should not need to have functions stored in it.